addon/build-waiter.ts

```diff
@@ -3,6 +3,9 @@ import { Primitive, TestWaiter, TestWaiterDebugInfo, WaiterName } from 'ember-te
 import { DEBUG } from '@glimmer/env';
 import Token from './token';
 import { register } from './waiter-manager';
+import { warn } from '@ember/debug';
+
+const WAITER_NAME_PATTERN = /^[^:]*:.*/;
```

Comment:
I think this needs to be tweaked a bit, to ensure the patterns listed in the README work nicely. Specifically, these two:

> 1. `file-name` - if your file has only one waiter
> 1. `addon-name` - if your addon has only one waiter

Neither of these have a `:`, and right now the regexp would cause them to warn.

Reply:
Yep. Good catch. Will add `?` for optional `:`.

Reply:
Updated.